python/pyspark/pandas/frame.py

```diff
@@ -425,42 +423,64 @@ class DataFrame(Frame, Generic[T]):
     def __init__(  # type: ignore[no-untyped-def]
```

Comment:
BTW, should we maybe add some examples above in the docstring?

Reply:
done